src/polyfills/template_polyfill.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ *
+ * A lightweight <template> polyfill that supports minimum features to cover
+ * lit-html use cases. It provides an alternate route in case <template> is not
+ * natively supported.
+ * Please note that nested template, cloning template node and innerHTML getter
+ * do NOT work with this polyfill.
+ * If it can not fullfill your requirement, please consider using the full
+ * polyfill: https://github.com/webcomponents/template
+ */
+export const initTemplatePolyfill = (forced = false) => {
+  if (typeof HTMLTemplateElement !== 'undefined' && !forced) {
+    return;
+  }
+  const contentDoc = document.implementation.createHTMLDocument('template');
+
+  const upgrade = (template: any) => {
+    template.content = contentDoc.createDocumentFragment();
+    defineInnerHTML(template);
+  };
+
+  const defineInnerHTML = (obj: any) => {
+    Object.defineProperty(obj, 'innerHTML', {
+      set: function(text) {
+        contentDoc.body.innerHTML = text;
+        const template = this as HTMLTemplateElement;
+        while (template.content.firstChild) {
+          template.content.removeChild(template.content.firstChild);
+        }
+        const body = contentDoc.body;
+        while (body.firstChild) {
+          template.content.appendChild(body.firstChild);
+        }
+      },
+      configurable: true
+    });
+  };
+
+  const capturedCreateElement = Document.prototype.createElement;
+  Document.prototype.createElement = function createElement() {
+    let el = capturedCreateElement.apply(this, arguments);
+    if (el.localName === 'template') {
+      el = capturedCreateElement.call(this, 'div');
```

Comment:
I forget: what''s the reason for not creating a "template" element? Can you document it here in a comment?

Reply:
The reason for using createElement('div') instead of createElement('template') is because deleting window.HTMLTemplateElement won't prevent document.createElement('template') from creating the real template element in chrome. Modifying template.content will then throw an exception and causing tests to fail.

I am glad you brought it up. Having the test affect the production code design doesn't feel right. I don't know if there is a test environment that only runs on browsers that do not support template. Otherwise we have to introduce this workaround or exclude this test.

Reply:
I think we can run these tests only on IE. They won't be run on community PRs, but will be on merges to master.